Schematics generates invalid effect in NGRX 11 w/ ng generate effect [module]

Create a project with @ngrx/scehmatics and @ngrx/effect

Generate a new module:

`ng generate module test`

Create a new feature with ngrx/schematics:

`ng generate feature TestFeature`

With default parameters it creates an invalid NGRX 11 effect:

```@Injectable()
export class TestFeatureEffects {


  loadTestFeatures$ = createEffect(() => {
    return this.actions$.pipe( 

      ofType(TestFeatureActions.loadTestFeatures),
      /** An EMPTY observable only emits completion. Replace with your own observable API request */
      concatMap(() => EMPTY)
    );
  });


  constructor(private actions$: Actions) {}

}
```

It dispatches an EMPTY, what causes :
```
    ERROR in src/app/test-module/effect-test-module/test/test-feature.effects.ts:14:36 - error TS2345: Argument of type '() => Observable<never>' is not assignable to parameter of type '() => never'.
      Type 'Observable<never>' is not assignable to type 'never'.
    
    14   loadTestFeatures$ = createEffect(() => {
                                          ~~~~~~~
    
```
According to the introduction of the new typechecks this is an error without specifying `{dispatch: false}` in the EffectConfig.

## Expected behavior:
It creates a new effect with `{dispatch: false}`.

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):
NGRX 11.0.0 / @ngrx/schematics 11.0.0. / Angular 11.1.2

I believe it was introduced due to the stricter type-checks in NGRX 11, should work with pre-11 versions.

--- a/modules/schematics/src/effect/files/__name@dasherize@if-flat__/__name@dasherize__.effects.ts.template
+++ b/modules/schematics/src/effect/files/__name@dasherize@if-flat__/__name@dasherize__.effects.ts.template
@@ -1,12 +1,12 @@
 import { Injectable } from '@angular/core';
 import { Actions, <%= effectMethod %><% if (feature) { %>, ofType<% } %> } from '@ngrx/effects';
 <% if (feature && api) { %>import { catchError, map, concatMap } from 'rxjs/operators';
-import { EMPTY, of } from 'rxjs';
+import { Observable, EMPTY, of } from 'rxjs';
 <% if (!creators) {%>import { Load<%= classify(name) %>sFailure, Load<%= classify(name) %>sSuccess, <%= classify(name) %>ActionTypes, <%= classify(name) %>Actions } from '<%= featurePath(group, flat, "actions", dasherize(name)) %><%= dasherize(name) %>.actions';<% } %>
 <% if (creators) {%>import * as <%= classify(name) %>Actions from '<%= featurePath(group, flat, "actions", dasherize(name)) %><%= dasherize(name) %>.actions';<% } %>
 <% } %>
 <% if (feature && !api) { %>import { concatMap } from 'rxjs/operators';
-import { EMPTY } from 'rxjs';
+import { Observable, EMPTY } from 'rxjs';
 <% if (!creators) {%>import { <%= classify(name) %>ActionTypes, <%= classify(name) %>Actions } from '<%= featurePath(group, flat, "actions", dasherize(name)) %><%= dasherize(name) %>.actions';<% } %>
 <% if (creators) {%>import * as <%= classify(name) %>Actions from '<%= featurePath(group, flat, "actions", dasherize(name)) %><%= dasherize(name) %>.actions';<% } %>
 <% } %>
@@ -38,13 +38,13 @@ export class <%= classify(name) %>Effects {
   <%= effectStart %>
     ofType(<%= classify(name) %>ActionTypes.Load<%= classify(name) %>s),
     /** An EMPTY observable only emits completion. Replace with your own observable API request */
-    concatMap(() => EMPTY)
+    concatMap(() => EMPTY as Observable<{ type: string }>)
   <%= effectEnd %>
 <% } else if (feature && !api && creators) { %>
   <%= effectStart %>
       ofType(<%= classify(name) %>Actions.load<%= classify(name) %>s),
       /** An EMPTY observable only emits completion. Replace with your own observable API request */
-      concatMap(() => EMPTY)
+      concatMap(() => EMPTY as Observable<{ type: string }>)
   <%= effectEnd %>
 <% } %>
 <% if (feature && !creators) { %>

--- a/modules/schematics/src/effect/index.spec.ts
+++ b/modules/schematics/src/effect/index.spec.ts
@@ -321,14 +321,16 @@ describe('Effect Schematic', () => {
       /import { Actions, Effect, ofType } from '@ngrx\/effects';/
     );
     expect(content).toMatch(/import { concatMap } from 'rxjs\/operators';/);
-    expect(content).toMatch(/import { EMPTY } from 'rxjs';/);
+    expect(content).toMatch(/import { Observable, EMPTY } from 'rxjs';/);
     expect(content).toMatch(
       /import { FooActionTypes, FooActions } from '\.\/foo.actions';/
     );
     expect(content).toMatch(/export class FooEffects/);
     expect(content).toMatch(/loadFoos\$ = this\.actions\$.pipe\(/);
     expect(content).toMatch(/ofType\(FooActionTypes\.LoadFoos\)/);
-    expect(content).toMatch(/concatMap\(\(\) => EMPTY\)/);
+    expect(content).toMatch(
+      /concatMap\(\(\) => EMPTY as Observable<\{ type: string \}>\)/
+    );
 
     expect(content).toMatch(
       /constructor\(private actions\$: Actions<FooActions>\) {}/
@@ -371,7 +373,7 @@ describe('Effect Schematic', () => {
     expect(content).toMatch(
       /import { catchError, map, concatMap } from 'rxjs\/operators';/
     );
-    expect(content).toMatch(/import { EMPTY, of } from 'rxjs';/);
+    expect(content).toMatch(/import { Observable, EMPTY, of } from 'rxjs';/);
     expect(content).toMatch(
       /import { LoadFoosFailure, LoadFoosSuccess, FooActionTypes, FooActions } from '\.\/foo.actions';/
     );

--- a/modules/schematics/src/feature/index.spec.ts
+++ b/modules/schematics/src/feature/index.spec.ts
@@ -229,7 +229,9 @@ describe('Feature Schematic', () => {
     expect(fileContent).toMatch(
       /import { catchError, map, concatMap } from 'rxjs\/operators';/
     );
-    expect(fileContent).toMatch(/import { EMPTY, of } from 'rxjs';/);
+    expect(fileContent).toMatch(
+      /import { Observable, EMPTY, of } from 'rxjs';/
+    );
     expect(fileContent).toMatch(
       /import \* as FooActions from '.\/foo.actions';/
     );

